src/lua/error.c

```diff
@@ -93,17 +93,46 @@ luaT_pusherror(struct lua_State *L, struct error *e)
 	luaL_setcdatagc(L, -2);
 }
 
+void
+luaT_error_set_trace(lua_State *L, int level, struct error *error)
+{
+	const char *file = "";
+	unsigned line = 0;
+	lua_Debug info;
+	if (level > 0 &&
+	    lua_getstack(L, level, &info) &&
+	    lua_getinfo(L, "Sl", &info)) {
+		if (*info.short_src) {
+			file = info.short_src;
+		} else if (*info.source) {
+			file = info.source;
+		} else {
+			file = "eval";
+		}
+		line = info.currentline;
+	}
+	error_set_location(error, file, line);
+}
+
 int
-luaT_error(lua_State *L)
+luaT_error_trace(lua_State *L, int level)
```

Comment:
Maybe, better call it `luaT_error_at`? Like `utils.call_at`.

Reply:
Done.